example/lib/location_flow/repository/location_repository.dart

```diff
@@ -9,25 +9,27 @@ class LocationRepository {
   Future<List<State>> getStates(String country) async {
     await _wait();
     final countryList = await countries();
-    final selectedCountry = countryList.firstWhere(
-      (element) => element.name == country,
-      orElse: () => null,
-    );
-    if (selectedCountry == null) return [];
-    final stateList = await states();
-    return stateList.where((s) => s.countryId == selectedCountry.id).toList();
+    try {
+      final selectedCountry =
+          countryList.firstWhere((element) => element.name == country);
+      final stateList = await states();
+      return stateList.where((s) => s.countryId == selectedCountry.id).toList();
+    } on StateError catch (_) {
```

Comment:
I don't think we should catch a `StateError` since error objects represent program failures that the programmer should have avoided. Can we instead revert to the previous implementation with the `orElse: () => null`?